src/column-layout/with-css.tsx

```diff
@@ -0,0 +1,86 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+import clsx from 'clsx';
+import flattenChildren from 'react-keyed-flatten-children';
+import { useContainerQuery } from '../internal/hooks/container-queries';
+import { getCommonClasses } from './util';
+import { InternalColumnLayoutProps } from './interfaces';
+import customCssProps from '../internal/generated/custom-css-properties';
+import styles from './styles.css.js';
+
+const isOdd = (value: number): boolean => value % 2 !== 0;
+
+export function calculcateCssColumnCount(
+  columns: number,
+  minColumnWidth: number,
+  containerWidth: number | null
+): number {
+  if (!containerWidth) {
+    return columns;
+  }
+
+  // First, calculate how many columns we can have based on the current container width and minColumnWidth.
+  const targetColumnCount = Math.min(columns, Math.floor(containerWidth / minColumnWidth));
+
+  // When we start wrapping into fewer columns than desired, we want to keep the number of columns even.
```

Comment:
Why do we want to keep the number even? It may still waste some space.

Reply:
The reason is to keep the behavior more aligned with the default Column layout.

Reply:
I wonder if it still makes sense if `minColumnWidth` is set. It feel it is implicitly taking even number of columns as higher priority than minColumnWidth. Although I think `minColumnWidth` should help to take as much space as possible.

Reply:
As discussed offline, it makes sense to have balanced view.